core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -607,6 +670,39 @@ private ExprValue exprAddDateDays(ExprValue date, ExprValue days) {
         : exprValue);
   }
 
+  /**
+   * Adds or subtracts time to/from date and returns the result.
+   *
+   * @param date A Date/Time/Datetime/Timestamp value to change.
+   * @param time A Date/Time/Datetime/Timestamp object to add/subtract time from.
+   * @param add A flag: true to add, false to subtract.
+   * @return A value calculated.
+   */
+  private ExprValue exprApplyTime(ExprValue date, ExprValue time, Boolean add) {
+    var timeToAdd = time.type() == TIME
+        ? time.timeValue()
```

Comment:
could we add the logic in the the ExprValue to pull time from datetimeValue() or dateValue() if timeValue() doesn't exist?

Reply:
FIxed in 61875c39.